impeller/entity/contents/vertices_contents.cc

```diff
@@ -30,62 +30,202 @@ void VerticesContents::SetGeometry(std::unique_ptr<VerticesGeometry> geometry) {
+    case BlendMode::kXor:
+      cmd.pipeline = renderer.GetAtlasBlendXorPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kPlus:
+      cmd.pipeline = renderer.GetAtlasBlendPlusPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kModulate:
+      cmd.pipeline = renderer.GetAtlasBlendModulatePipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    // Advanced
+    case BlendMode::kScreen:
+      cmd.pipeline = renderer.GetAtlasBlendScreenPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kOverlay:
+      cmd.pipeline = renderer.GetAtlasBlendOverlayPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kDarken:
+      cmd.pipeline = renderer.GetAtlasBlendDarkenPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kLighten:
+      cmd.pipeline = renderer.GetAtlasBlendLightenPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kColorDodge:
+      cmd.pipeline = renderer.GetAtlasBlendColorDodgePipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kColorBurn:
+      cmd.pipeline = renderer.GetAtlasBlendColorBurnPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kHardLight:
+      cmd.pipeline = renderer.GetAtlasBlendHardLightPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kSoftLight:
+      cmd.pipeline = renderer.GetAtlasBlendSoftLightPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kDifference:
+      cmd.pipeline = renderer.GetAtlasBlendDifferencePipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kExclusion:
+      cmd.pipeline = renderer.GetAtlasBlendExclusionPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kMultiply:
+      cmd.pipeline = renderer.GetAtlasBlendMultiplyPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kHue:
+      cmd.pipeline = renderer.GetAtlasBlendHuePipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kSaturation:
+      cmd.pipeline = renderer.GetAtlasBlendSaturationPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kColor:
+      cmd.pipeline = renderer.GetAtlasBlendColorPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
+    case BlendMode::kLuminosity:
+      cmd.pipeline = renderer.GetAtlasBlendLuminosityPipeline(
+          OptionsFromPassAndEntity(pass, entity));
+      break;
   }
-  pass.AddCommand(std::move(cmd));
+  auto src_texture = src_contents_->RenderToSnapshot(renderer, entity);
```

Comment:
This almost works, or rather it works in the playground example I've set up but in a sample app the gradient starts/stops don't seem to be lined up correctly. I think I've probably either forgotten to apply a transform somewhere, or else I'm double transforming - that said, where. 😓